src/main/java/edu/harvard/iq/dataverse/authorization/providers/oauth2/AbstractOAuth2AuthenticationProvider.java

```diff
@@ -207,6 +214,14 @@ public AuthenticationProviderDisplayInfo getInfo() {
         return new AuthenticationProviderDisplayInfo(getId(), getTitle(), getSubTitle());
     }
     
+    /**
+     * Used in {@link LoginPage#listAuthenticationProviders()} for sorting the providers in the UI
+     * TODO: this might be extended to use a value set by the admin when configuring the provider via JSON.
+     * @return an integer value (sort ascending)
+     */
+    @Override
+    public int getOrder() { return 100; }
```

Comment:
What if there are more than 100 authentication providers? Some have the same number? Is it ok to have two 42s?

Reply:
Actually it doesn't matter if some have the same number, as they are sorted by number first, then by name.

This is mostly about sorting Builtin first (order = 1), Shib second (order=20) and OAuth last (all order = 100), as it has been sorted before my refactoring.

Once we have a configurable value for this, you could also realize sth. like provider A first, then all the others.
[Please see also my comment above.](https://github.com/IQSS/dataverse/pull/6433/files/511827429f280ee7ad997fa68fc85694a556640d?file-filters%5B%5D=.java&file-filters%5B%5D=.properties#r354178859)